csharp/src/Ice/IceSSL/SSLEngine.cs

```diff
@@ -260,12 +228,68 @@ internal void verifyPeer(ConnectionInfo info, string description)
         }
     }
 
+    internal SslClientAuthenticationOptions clientAuthenticationOptions(
+        RemoteCertificateValidationCallback remoteCertificateValidationCallback,
+        string host)
+    {
+        return new SslClientAuthenticationOptions
+        {
+            ClientCertificates = _certs,
+            LocalCertificateSelectionCallback = (sender, targetHost, certs, remoteCertificate, acceptableIssuers) =>
```

Comment:
This was previously done in https://github.com/zeroc-ice/ice/blob/3ca96ffbe6a56660232cf2942a7238ac90babadb/csharp/src/Ice/IceSSL/TransceiverI.cs#L515

Is still the same logic.